File: Dictionaries and Sets/1. Two Sum/1. Two Sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let seen = new Map()
    for (let i = 0; i < nums.length; i++) {
        if (seen.has(nums[i])) {
            return [seen.get(nums[i]), i]
        }
        seen.set(target - nums[i], i)
    }
};

// Time: O(n)
// Space: O(1)

File: Dictionaries and Sets/1. Two Sum/1. Two Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = {}
        for i, num in enumerate(nums):
            if num in seen:
                return [seen[num], i]
            seen[target - num] = i

# Time: O(n)
# Space: O(n)